### PR TITLE
Demo_Kekkai, its object, and its cutscenes

### PR DESCRIPTION
**Pull request by petrie911**
Demo_Kekkai is the barriers in ganon's castle, while its object is everything in ganon's castle. Updated all relevant actors to use the new variable names (including two undecomped ones) and z_demo to use the cutscene names.

Merge will need to wait on vertex array support from ZAPD.

### CODE REVIEW

File: src/overlays/actors/ovl_Demo_Kekkai/z_demo_kekkai.c [hunk 1]
```diff
@@ -47,23 +54,285 @@ static ColliderCylinderInit D_8098E0B0 = {
...
+}
+
+void DemoKekkai_Init(Actor* thisx, GlobalContext* globalCtx) {
+    s32 pad;
+    DemoKekkai* this = THIS;
+
+    this->sfxFlag = 0;
+    this->energyAlpha = 1.0f;
+    Actor_SetScale(thisx, 0.1f);
+    thisx->colChkInfo.mass = 0xFF;
```
Comment (fig02) [inline src/overlays/actors/ovl_Demo_Kekkai/z_demo_kekkai.c:83]: mass define

**Review by fig02 — COMMENTED** [review]
just one thing

**Review by fig02 — APPROVED** [review]
changes made

File: src/overlays/actors/ovl_Bg_Gnd_Firemeiro/z_bg_gnd_firemeiro.c [hunk 2]
```diff
@@ -19,8 +20,8 @@ void func_808795AC(BgGndFiremeiro* this, GlobalContext* globalCtx);
 void func_80879668(BgGndFiremeiro* this, GlobalContext* globalCtx);
 void func_80879808(BgGndFiremeiro* this, GlobalContext* globalCtx);
 
-extern UNK_TYPE D_0600E210;
-extern UNK_TYPE D_0600ECD8;
+extern Gfx gFireTrialPlatformDL[];
+extern CollisionHeader gFireTrialPlatformCol;
```
Comment (Roman971) [inline src/overlays/actors/ovl_Bg_Gnd_Firemeiro/z_bg_gnd_firemeiro.c:24]: Why is the object header not used here?

File: src/overlays/actors/ovl_Object_Kankyo/z_object_kankyo.c [hunk 3]
```diff
@@ -17,7 +18,7 @@ void ObjectKankyo_Update(Actor* thisx, GlobalContext* globalCtx);
 void ObjectKankyo_Draw(Actor* thisx, GlobalContext* globalCtx);
 
 extern UNK_TYPE D_01000000;
-extern UNK_TYPE D_06005FF0;
+extern Gfx gDemoKekkaiDL_005FF0[];
```
Comment (Roman971) [inline src/overlays/actors/ovl_Object_Kankyo/z_object_kankyo.c:21]:
```suggestion
```

File: undefined_syms.txt [hunk 4]
```diff
@@ -2007,7 +2007,7 @@ D_06008390 = 0x06008390;
 // z_demo_kekkai
 D_06004930 = 0x06004930;
 D_06004F00 = 0x06004F00;
-D_06004FD0 = 0x06004FD0;
+gTrialBarrierEnergyVtx = 0x06004FD0;
```
Comment (Roman971) [inline undefined_syms.txt:2010]:
```suggestion
```

File: src/overlays/actors/ovl_Bg_Gnd_Firemeiro/z_bg_gnd_firemeiro.c (hunk 2 above)
Comment (petrie911) [reply src/overlays/actors/ovl_Bg_Gnd_Firemeiro/z_bg_gnd_firemeiro.c:24]: It gave some error when I tried using the header, though unfortunately I don't remember what.

Comment (fig02) [reply src/overlays/actors/ovl_Bg_Gnd_Firemeiro/z_bg_gnd_firemeiro.c:24]: not sure what the issue was but its fine now

**Comment by fig02** [comment]
changes made

**Review by Roman971 — APPROVED** [review]

**Pull request merged**